### PR TITLE
[ONL-6712] - migrate ec-timer to vue3

### PR DESCRIPTION
**Pull request by DisegnAle**
migrated time componet to vue3 
Paired with @derrick-g

### CODE REVIEW

File: src/components/ec-timer/ec-timer.vue [hunk 1]
```diff
@@ -46,96 +46,87 @@
...
+      return Number.isInteger(value) && value > 0;
     },
   },
-  emits: ['time-expired'],
-  data() {
-    return {
-      radius: 24,
-      strokeWidth: 4,
-      secondsLeft: this.seconds,
-      startTime: null,
```
Comment (DisegnAle) [inline src/components/ec-timer/ec-timer.vue:79]: this variable as well as currentTime were not used. We removed them

File: src/components/ec-timer/ec-timer.vue [hunk 2]
```diff
@@ -46,96 +46,87 @@
...
+const diameter = computed(() => radius.value * 2 + strokeWidth.value);
+const viewbox = computed(() => `0 0 ${diameter.value} ${diameter.value}`);
+const circumference = computed(() => 2 * Math.PI * radius.value);
+const steps = computed(() => circumference.value / props.seconds);
+const offset = computed(() => circumference.value + steps.value * secondsLeft.value);
+
+const radius = ref(24);
+const strokeWidth = ref(4);
+const secondsLeft = ref(props.seconds);
+const countdown = ref(null);
```
Comment (mcibique) [inline src/components/ec-timer/ec-timer.vue:86]: Do we need the countdown to be reactive? is it used anywhere in the template or computed props? We can save some work for Vue's reactivity and just do this without `ref`. We don't need to then `countdown.value` anywhere but just `countdown`.

Comment (derrick-g) [reply src/components/ec-timer/ec-timer.vue:86]: Agreed, it can just be a basic const. As its not in the template nor a computed function.

Comment (DisegnAle) [reply src/components/ec-timer/ec-timer.vue]: changed it to be a let

**Review by mcibique — APPROVED** [review]

**Review by jeger-at — APPROVED** [review]

**Pull request merged**